server/mbta_api.py

```diff
@@ -110,11 +160,21 @@ async def vehicle_data_for_routes(route_ids, test_mode=False):
                     "isNewTrain": is_new,
                 }
             )
-        except (KeyError, TypeError, AttributeError):
+        except Exception as e:
+            eastern = pytz.timezone("US/Eastern")
+            now_eastern = datetime.datetime.now(eastern)
+            print(f"[{now_eastern}] Error processing vehicle {vehicle}: {e}")
             continue
-    return vehicles_to_display
 
+    vehicle_stats["green_percent_active_new"] = percent(vehicle_stats["green_new_active"], vehicle_stats["green_total_active"])
+    vehicle_stats["orange_percent_active_new"] = percent(vehicle_stats["orange_new_active"], vehicle_stats["orange_total_active"])
+    vehicle_stats["red_percent_active_new"] = percent(vehicle_stats["red_new_active"], vehicle_stats["red_total_active"])
```

Comment:
maybe we could leave this for the client to calculate?

Reply:
So the only hitch with doing client-side calculations is we are currently only sending new trains to the client, so we lose data pertaining to old trains (i.e. total active train counts). We _could_ send all trains to the frontend and do both the stat and "vehicles to display" calculations there, but I feel like we're better off doing the computation server-side and only sending data that's needed to the client.

Reply:
ah, i meant just the percent calculation (division). Assuming we're sending new counts and total counts.

Reply:
Ahhhh I see - suppose we could do it either way in that case. Any specific advantages/disadvantages to either? @idreyn

Reply:
I agree that we should do division on the client.